### PR TITLE
Bugfix: Fixes issue where focusing on editor always placed caret at offset 0

### PR DESCRIPTION
**Pull request by kaydelaney**
#### Is this adding or improving a _feature_ or fixing a _bug_?
Fixing a bug.

#### What's the new behavior?
Focusing on editor places caret next to cursor as expected.

#### How does this change work?
Previously, in Firefox, focusing on an editor would result in `updateSelection` being called before
`onNativeSelectionChange` is triggered, but after the native selection has been changed, resulting in `removeAllRanges` being called on the native selection. To prevent this, this PR introduces a member `nativeSelection` which stores the native selection's state when `onNativeSelectionChange` is triggered, so that when `updateSelection` is called, we can check if the native selection has been updated before `onNativeSelectionChange` has been triggered, and if so, prevent `removeAllRanges` from being called.

#### Have you checked that...?

* [x] The new code matches the existing patterns and styles.
* [x] The tests pass with `yarn test`.
* [x] The linter passes with `yarn lint`. (Fix errors with `yarn prettier`.)
* [x] The relevant examples still work. (Run examples with `yarn watch`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #2978

### CODE REVIEW

**Comment by joechesney** [comment]
lgtm

File: packages/slate-react/src/components/content.js [hunk 1]
```diff
@@ -78,6 +98,8 @@ class Content extends React.Component {
     tagName: 'div',
   }
 
+  nativeSelection = {}
```
Comment (ianstormtaylor) [inline packages/slate-react/src/components/content.js]: Can you add a comment here explaining what this is for?

File: packages/slate-react/src/components/content.js [hunk 2]
```diff
@@ -237,7 +259,12 @@ class Content extends React.Component {
 
     // If the Slate selection is unset, but the DOM selection has a range
     // selected in the editor, we need to remove the range.
-    if (selection.isUnset && rangeCount && this.isInEditor(anchorNode)) {
+    if (
+      selection.isUnset &&
+      rangeCount &&
+      this.isInEditor(anchorNode) &&
+      selectionsEqual(this.nativeSelection, native)
```
Comment (ianstormtaylor) [inline packages/slate-react/src/components/content.js]: Since this is only used once, can you inline this logic?

Comment (ianstormtaylor) [reply packages/slate-react/src/components/content.js]: Also, can you add a comment to explain what is going on there by comparing the current native selection to a cached value?

File: packages/slate-react/src/components/content.js (hunk 1 above)
Comment (ianstormtaylor) [reply packages/slate-react/src/components/content.js]: Also, it seems like this might be better placed in `this.tmp.selection`?

**Comment by Ridd0** [comment]
@ianstormtaylor Is there any chance to merge it in this week? Thanks!

**Comment by ianstormtaylor** [comment]
Ah sorry missed the review request. Looks good to me, thanks so much @kaydelaney!

**Pull request merged**